bugsnag-plugin-android-ndk/src/main/jni/utils/string.c

```diff
@@ -1,40 +1,14 @@
 #include "string.h"
-#include <limits.h>
-#include <stdbool.h>
 #include <string.h>
 
-void bsg_strcpy(char *dst, char *src) { bsg_strncpy(dst, src, INT_MAX); }
-
-void bsg_strncpy(char *dst, char *src, size_t len) {
-  int i = 0;
-  while (i <= len) {
-    char current = src[i];
-    dst[i] = current;
-    if (current == '\0') {
-      break;
-    }
-    i++;
-  }
-}
-
-size_t bsg_strlen(const char *str) {
-  size_t i = 0;
-  while (true) {
-    char current = str[i];
-    if (current == '\0') {
-      return i;
-    } else if (i == INT_MAX) {
-      return INT_MAX;
+char *bsg_strncpy_safe(char *dst, const char *src, size_t dst_size) {
```

Comment:
Perhaps overly paranoid, but what will happen here if `dst` is `NULL`?

Reply:
It crashes, like any copy function would :)

It's a dangerous path once you start putting guardrails over function params, as it masks out coding errors and pushes them further and further from the source of the problem. I'm even nervous about the guard on `src`, but it's old code so I didn't touch it.

Reply:
Happy enough to leave this as is - I haven't spotted anywhere where we are passing it `NULL`, and it's something we can control so could fix if it ever did go wrong.